java/compaction/compaction-job-execution/src/main/java/sleeper/compaction/job/execution/JavaCompactionRunner.java

```diff
@@ -144,14 +145,11 @@ public static CloseableIterator<Record> getMergingIterator(
         if (null != compactionJob.getIteratorClassName()) {
             SortedRecordIterator iterator;
             try {
-                iterator = objectFactory.getObject(compactionJob.getIteratorClassName(), SortedRecordIterator.class);
+                IteratorFactory iterFactory = new IteratorFactory(objectFactory);
+                iterator = iterFactory.getIterator(compactionJob.getIteratorClassName(), compactionJob.getIteratorConfig(), schema);
             } catch (ObjectFactoryException e) {
                 throw new IteratorCreationException("ObjectFactoryException creating iterator of class " + compactionJob.getIteratorClassName(), e);
```

Comment:
I think it seems cleaner to throw the IteratorCreationException from inside the IteratorFactory. That would be a change for the Athena module, but that's experimental to begin with and it seems like a less descriptive exception to throw ObjectFactoryException directly.